Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RnwServiceTest.cs

```diff
@@ -0,0 +1,117 @@
+            _fakeRnwRepository = A.Fake<IRnwRepository>();
+
+            _rnwService = new RnwService(_fakeRnwRepository, _fakeLogger);
+        }
+
+        [Test]
+        public async Task WhenRequestToGetWarningTypes_ThenReturnListOfWarningType()
+        {
+            DateTime _fakeDateTime = DateTime.UtcNow;
+            _fakeRadioNavigationalWarnings.DateTimeGroup = _fakeDateTime;
+
+            A.CallTo(() => _fakeRnwRepository.GetWarningTypes()).Returns(new List<WarningType>() { new WarningType {Id=1, Name="test"}});
+
+            List<WarningType> result = await _rnwService.GetWarningTypes();
+
+            Assert.AreEqual("test", result[0].Name);
+        }
+
+        [Test]
+        public async Task WhenPostValidRequest_ThenReturnTrue()
+        {
+            DateTime _fakeDateTime = DateTime.UtcNow;
+            _fakeRadioNavigationalWarnings.DateTimeGroup = _fakeDateTime;
+
+            bool result = await _rnwService.CreateNewRadioNavigationWarningsRecord(_fakeRadioNavigationalWarnings, CorrelationId);
+
+            Assert.IsTrue(result);
+        }
+
+        [Test]
+        public void WhenPostInvalidWarningTypeInRequest_ThenReturnFalse()
+        {
+            DateTime _fakeDateTime = DateTime.UtcNow;
+            _fakeRadioNavigationalWarnings.DateTimeGroup = _fakeDateTime;
+            _fakeRadioNavigationalWarnings.WarningType = 3;
+
+            Assert.ThrowsAsync(Is.TypeOf<InvalidDataException>().And.Message.EqualTo("Invalid value recieved for parameter warningType"),
+                             async delegate { await _rnwService.CreateNewRadioNavigationWarningsRecord(_fakeRadioNavigationalWarnings, CorrelationId); });
+        }
+
+        [Test]
+        public void WhenPostInvalidReferenceInRequest_ThenReturnFalse()
+        {
+            DateTime _fakeDateTime = DateTime.UtcNow;
+            _fakeRadioNavigationalWarnings.DateTimeGroup = _fakeDateTime;
+            _fakeRadioNavigationalWarnings.Reference = "";
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>().And.Message.EqualTo("Invalid value recieved for parameter reference"),
+                             async delegate { await _rnwService.CreateNewRadioNavigationWarningsRecord(_fakeRadioNavigationalWarnings, CorrelationId); });
+        }
+
+        [Test]
+        public void WhenPostInvalidSummaryInRequest_ThenReturnFalse()
+        {
+            DateTime _fakeDateTime = DateTime.UtcNow;
+            _fakeRadioNavigationalWarnings.DateTimeGroup = _fakeDateTime;
+            _fakeRadioNavigationalWarnings.Summary = "";
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>().And.Message.EqualTo("Invalid value recieved for parameter summary"),
+                             async delegate { await _rnwService.CreateNewRadioNavigationWarningsRecord(_fakeRadioNavigationalWarnings, CorrelationId); });
+        }
+
+        [Test]
+        public void WhenPostInvalidContentInRequest_ThenReturnFalse()
```

Comment:
Method incorrectly named, as the test case checks for an exception thrown (not a return value of false)

Reply:
done required changes.